Use ASCIILiteral in more places

cdumez opened the pull request:
#### 918bd357adca5b0a818bdb200e4ce5a74f292b75
<pre>
Use ASCIILiteral in more places
<a href="https://bugs.webkit.org/show_bug.cgi?id=273111">https://bugs.webkit.org/show_bug.cgi?id=273111</a>

Reviewed by Darin Adler.

Use ASCIILiteral in more places to reduce use of `const char*`.

* Source/WebCore/html/parser/HTMLTokenizer.cpp:
(WebCore::HTMLTokenizer::processToken):
* Source/WebCore/loader/archive/mhtml/MHTMLArchive.cpp:
(WebCore::MHTMLArchive::generateMHTMLData):
* Source/WebCore/page/History.cpp:
(WebCore::History::stateObjectAdded):
* Source/WebCore/page/MemoryRelease.cpp:
(WebCore::logMemoryStatisticsReasonDescription):
(WebCore::logMemoryStatistics):
* Source/WebCore/page/PageOverlayController.cpp:
(WebCore::PageOverlayController::shouldDumpPropertyForLayer const):
* Source/WebCore/page/PageOverlayController.h:
* Source/WebCore/page/PerformanceMonitor.cpp:
(WebCore::stringForCPUSamplingActivityState):
(WebCore::PerformanceMonitor::measureCPUUsageInActivityState):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::consoleMessageForViolation):
(WebCore::ContentSecurityPolicy::allowJavaScriptURLs const):
(WebCore::ContentSecurityPolicy::allowInlineEventHandlers const):
(WebCore::ContentSecurityPolicy::allowNonParserInsertedScripts const):
(WebCore::ContentSecurityPolicy::allowInlineScript const):
(WebCore::ContentSecurityPolicy::allowInlineStyle const):
(WebCore::ContentSecurityPolicy::allowEval const):
(WebCore::ContentSecurityPolicy::allowFrameAncestors const):
(WebCore::ContentSecurityPolicy::allowPluginType const):
(WebCore::ContentSecurityPolicy::allowObjectFromSource const):
(WebCore::ContentSecurityPolicy::allowChildFrameFromSource const):
(WebCore::ContentSecurityPolicy::allowResourceFromSource const):
(WebCore::ContentSecurityPolicy::allowWorkerFromSource const):
(WebCore::ContentSecurityPolicy::allowScriptFromSource const):
(WebCore::ContentSecurityPolicy::allowStyleFromSource const):
(WebCore::ContentSecurityPolicy::allowConnectToSource const):
(WebCore::ContentSecurityPolicy::allowBaseURI const):
* Source/WebCore/platform/cocoa/PasteboardCocoa.mm:
(WebCore::imageTypeToMIMEType):
(WebCore::Pasteboard::read):
* Source/WebCore/platform/gamepad/PlatformGamepad.h:
(WebCore::PlatformGamepad::source const):
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.h:
* Source/WebCore/platform/gamepad/mac/HIDGamepad.h:
* Source/WebCore/platform/gamepad/mac/MultiGamepadProvider.h:
* Source/WebCore/platform/graphics/GraphicsLayer.cpp:
(WebCore::GraphicsLayer::dumpProperties const):
* Source/WebCore/platform/graphics/GraphicsLayerClient.h:
(WebCore::GraphicsLayerClient::shouldDumpPropertyForLayer const):
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::removeSamplesFromTrackBuffer):
(WebCore::SourceBufferPrivate::processMediaSample):
* Source/WebCore/platform/graphics/TrackBuffer.cpp:
(WebCore::TrackBuffer::removeSamples):
(WebCore::TrackBuffer::removeCodedFrames):
* Source/WebCore/platform/graphics/TrackBuffer.h:
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::GraphicsLayerCA::updateCoverage):
(WebCore::GraphicsLayerCA::purposeNameForInnerLayer const):
(WebCore::GraphicsLayerCA::dumpInnerLayer const):
(WebCore::GraphicsLayerCA::dumpAnimations):
(WebCore::GraphicsLayerCA::layerChangeAsString):
(WebCore::GraphicsLayerCA::dumpLayerChangeFlags):
(WebCore::GraphicsLayerCA::dumpAdditionalProperties const):
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.h:
* Source/WebCore/platform/graphics/cocoa/WebMAudioUtilitiesCocoa.mm:
(WebCore::registerDecoderFactory):
(WebCore::registerOpusDecoderIfNeeded):
(WebCore::registerVorbisDecoderIfNeeded):
* Source/WebCore/platform/mac/PlatformPasteboardMac.mm:
(WebCore::safeTypeForDOMToReadAndWriteForPlatformType):
(WebCore::PlatformPasteboard::typesSafeForDOMToReadAndWrite const):
* Source/WebCore/platform/mock/MockRealtimeVideoSource.cpp:
(WebCore::MockRealtimeVideoSource::drawText):
* Source/WebCore/platform/mock/ScrollbarsControllerMock.cpp:
(WebCore::ScrollbarsControllerMock::scrollbarPrefix const):
(WebCore::ScrollbarsControllerMock::mouseEnteredScrollbar const):
(WebCore::ScrollbarsControllerMock::mouseExitedScrollbar const):
(WebCore::ScrollbarsControllerMock::mouseIsDownInScrollbar const):
* Source/WebCore/platform/mock/ScrollbarsControllerMock.h:
* Source/WebCore/platform/network/HTTPParsers.cpp:
(WebCore::skipToken):
(WebCore::parseXSSProtectionHeader):
* Source/WebCore/platform/text/SegmentedString.cpp:
(WebCore::SegmentedString::advancePastSlowCase):
* Source/WebCore/platform/text/SegmentedString.h:
(WebCore::SegmentedString::advancePast):
(WebCore::SegmentedString::advancePastLettersIgnoringASCIICase):
* Source/WebCore/rendering/LegacyInlineBox.cpp:
(WebCore::LegacyInlineBox::boxName const):
(WebCore::LegacyInlineBox::outputLineBox const):
* Source/WebCore/rendering/LegacyInlineBox.h:
* Source/WebCore/rendering/LegacyInlineFlowBox.cpp:
(WebCore::LegacyInlineFlowBox::boxName const):
* Source/WebCore/rendering/LegacyInlineFlowBox.h:
* Source/WebCore/rendering/LegacyInlineTextBox.cpp:
(WebCore::LegacyInlineTextBox::boxName const):
(WebCore::LegacyInlineTextBox::outputLineBox const):
* Source/WebCore/rendering/LegacyInlineTextBox.h:
* Source/WebCore/rendering/LegacyRootInlineBox.cpp:
(WebCore::LegacyRootInlineBox::boxName const):
* Source/WebCore/rendering/LegacyRootInlineBox.h:
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::outputPaintOrderTreeRecursive):
(WebCore::showPaintOrderTree):
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::shouldDumpPropertyForLayer const):
* Source/WebCore/rendering/RenderLayerBacking.h:
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::shouldDumpPropertyForLayer const):
(WebCore::RenderLayerCompositor::updateBackingAndHierarchy):
(WebCore::RenderLayerCompositor::logLayerInfo):
(WebCore::compositingReasonToString):
(WebCore::RenderLayerCompositor::logOneReasonForCompositing):
* Source/WebCore/rendering/RenderLayerCompositor.h:
* Source/WebCore/rendering/RenderQuote.cpp:
(WebCore::quotesForLanguage):
* Source/WebCore/rendering/svg/SVGRenderTreeAsText.cpp:
(WebCore::operator&lt;&lt;):
(WebCore::writeNameValuePair):
(WebCore::writeNameAndQuotedValue):
(WebCore::writeIfNotEmpty):
(WebCore::writeIfNotDefault):
(WebCore::writeSVGPaintingResource):
(WebCore::writeSVGFillPaintingResource):
(WebCore::writeSVGStrokePaintingResource):
(WebCore::writeSVGPaintingFeatures):
(WebCore::writeSVGGraphicsElement):
(WebCore::writeRenderSVGTextBox):
(WebCore::writeSVGInlineTextBox):
(WebCore::writeStandardPrefix):
(WebCore::writeCommonGradientProperties):
(WebCore::writeSVGResourceContainer):
(WebCore::writeSVGContainer):
(WebCore::write):
(WebCore::writeSVGText):
(WebCore::writeSVGInlineText):
(WebCore::writeSVGImage):
(WebCore::writeSVGGradientStop):
(WebCore::writeResources):

Canonical link: <a href="https://commits.webkit.org/277913@main">https://commits.webkit.org/277913@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8737c1c82cadeeaa9d2375109385c32cee8acc70

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48927 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28140 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51894 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51614 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44997 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/51232 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34092 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25668 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/39995 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/49696 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25784 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42183 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21102 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23244 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/43355 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6982 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/45175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/43857 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53525 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23978 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20237 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/47303 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25241 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42392 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/46259 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26050 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6995 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24961 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/loader/archive/mhtml/MHTMLArchive.cpp:
```diff
@@ -57,8 +57,8 @@
 
-const char* const quotedPrintable = "quoted-printable";
-const char* const base64 = "base64";
+static constexpr auto quotedPrintable = "quoted-printable"_s;
```
Pretty sure we don’t need `static` if we are using `constexpr` at the namespace level like this.

darinadler commented on Source/WebCore/platform/graphics/GraphicsLayer.cpp:
```diff
@@ -903,144 +903,144 @@ void GraphicsLayer::dumpProperties(TextStream& ts, OptionSet<LayerTreeAsTextOpti
-        ts << "[" << m_childrenTransform->m31() << " " << m_childrenTransform->m32() << " " << m_childrenTransform->m33() << " " << m_childrenTransform->m34() << "] ";
-        ts << "[" << m_childrenTransform->m41() << " " << m_childrenTransform->m42() << " " << m_childrenTransform->m43() << " " << m_childrenTransform->m44() << "])\n";
+        ts << indent << "(childrenTransform "_s;
+        ts << "["_s << m_childrenTransform->m11() << " "_s << m_childrenTransform->m12() << " "_s << m_childrenTransform->m13() << " "_s << m_childrenTransform->m14() << "] "_s;
```
Can we use `‘[‘` and the like for the single character ones? Maybe a global regular expression search and replace for `<< "(?)"_s` to `<< ‘\1’`. Unless TextStream doesn’t do `char`.

cdumez replied:
Looks like it works. I'll fix.

darinadler commented on Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
```diff
@@ -2787,21 +2787,21 @@ void GraphicsLayerCA::updateCoverage(const CommitState& commitState)
 #if !LOG_DISABLED
     if (requiresBacking) {
-        auto reasonForBacking = [&]() -> const char* {
+        auto reasonForBacking = [&]() -> ASCIILiteral {
```
Can we omit the return type here?

darinadler commented on Source/WebCore/platform/mock/MockRealtimeVideoSource.cpp:
```diff
@@ -573,25 +573,25 @@ void MockRealtimeVideoSource::drawText(GraphicsContext& context)
         string = makeString("Preset size: ", captureSize.width(), " x ", captureSize.height());
         context.drawText(drawingState.statsFont(), TextRun((StringView(string))), statsLocation);
```
This has too many parentheses in `TextRun` and uses `StringView` for no obvious reason.

cdumez committed:
Use ASCIILiteral in more places
https://bugs.webkit.org/show_bug.cgi?id=273111

Reviewed by Darin Adler.

Use ASCIILiteral in more places to reduce use of `const char*`.

* Source/WebCore/html/parser/HTMLTokenizer.cpp:
(WebCore::HTMLTokenizer::processToken):
* Source/WebCore/loader/archive/mhtml/MHTMLArchive.cpp:
(WebCore::MHTMLArchive::generateMHTMLData):
* Source/WebCore/page/History.cpp:
(WebCore::History::stateObjectAdded):
* Source/WebCore/page/MemoryRelease.cpp:
(WebCore::logMemoryStatisticsReasonDescription):
(WebCore::logMemoryStatistics):
* Source/WebCore/page/PageOverlayController.cpp:
(WebCore::PageOverlayController::shouldDumpPropertyForLayer const):
* Source/WebCore/page/PageOverlayController.h:
* Source/WebCore/page/PerformanceMonitor.cpp:
(WebCore::stringForCPUSamplingActivityState):
(WebCore::PerformanceMonitor::measureCPUUsageInActivityState):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::consoleMessageForViolation):
(WebCore::ContentSecurityPolicy::allowJavaScriptURLs const):
(WebCore::ContentSecurityPolicy::allowInlineEventHandlers const):
(WebCore::ContentSecurityPolicy::allowNonParserInsertedScripts const):
(WebCore::ContentSecurityPolicy::allowInlineScript const):
(WebCore::ContentSecurityPolicy::allowInlineStyle const):
(WebCore::ContentSecurityPolicy::allowEval const):
(WebCore::ContentSecurityPolicy::allowFrameAncestors const):
(WebCore::ContentSecurityPolicy::allowPluginType const):
(WebCore::ContentSecurityPolicy::allowObjectFromSource const):
(WebCore::ContentSecurityPolicy::allowChildFrameFromSource const):
(WebCore::ContentSecurityPolicy::allowResourceFromSource const):
(WebCore::ContentSecurityPolicy::allowWorkerFromSource const):
(WebCore::ContentSecurityPolicy::allowScriptFromSource const):
(WebCore::ContentSecurityPolicy::allowStyleFromSource const):
(WebCore::ContentSecurityPolicy::allowConnectToSource const):
(WebCore::ContentSecurityPolicy::allowBaseURI const):
* Source/WebCore/platform/cocoa/PasteboardCocoa.mm:
(WebCore::imageTypeToMIMEType):
(WebCore::Pasteboard::read):
* Source/WebCore/platform/gamepad/PlatformGamepad.h:
(WebCore::PlatformGamepad::source const):
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.h:
* Source/WebCore/platform/gamepad/mac/HIDGamepad.h:
* Source/WebCore/platform/gamepad/mac/MultiGamepadProvider.h:
* Source/WebCore/platform/graphics/GraphicsLayer.cpp:
(WebCore::GraphicsLayer::dumpProperties const):
* Source/WebCore/platform/graphics/GraphicsLayerClient.h:
(WebCore::GraphicsLayerClient::shouldDumpPropertyForLayer const):
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::removeSamplesFromTrackBuffer):
(WebCore::SourceBufferPrivate::processMediaSample):
* Source/WebCore/platform/graphics/TrackBuffer.cpp:
(WebCore::TrackBuffer::removeSamples):
(WebCore::TrackBuffer::removeCodedFrames):
* Source/WebCore/platform/graphics/TrackBuffer.h:
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::GraphicsLayerCA::updateCoverage):
(WebCore::GraphicsLayerCA::purposeNameForInnerLayer const):
(WebCore::GraphicsLayerCA::dumpInnerLayer const):
(WebCore::GraphicsLayerCA::dumpAnimations):
(WebCore::GraphicsLayerCA::layerChangeAsString):
(WebCore::GraphicsLayerCA::dumpLayerChangeFlags):
(WebCore::GraphicsLayerCA::dumpAdditionalProperties const):
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.h:
* Source/WebCore/platform/graphics/cocoa/WebMAudioUtilitiesCocoa.mm:
(WebCore::registerDecoderFactory):
(WebCore::registerOpusDecoderIfNeeded):
(WebCore::registerVorbisDecoderIfNeeded):
* Source/WebCore/platform/mac/PlatformPasteboardMac.mm:
(WebCore::safeTypeForDOMToReadAndWriteForPlatformType):
(WebCore::PlatformPasteboard::typesSafeForDOMToReadAndWrite const):
* Source/WebCore/platform/mock/MockRealtimeVideoSource.cpp:
(WebCore::MockRealtimeVideoSource::drawText):
* Source/WebCore/platform/mock/ScrollbarsControllerMock.cpp:
(WebCore::ScrollbarsControllerMock::scrollbarPrefix const):
(WebCore::ScrollbarsControllerMock::mouseEnteredScrollbar const):
(WebCore::ScrollbarsControllerMock::mouseExitedScrollbar const):
(WebCore::ScrollbarsControllerMock::mouseIsDownInScrollbar const):
* Source/WebCore/platform/mock/ScrollbarsControllerMock.h:
* Source/WebCore/platform/network/HTTPParsers.cpp:
(WebCore::skipToken):
(WebCore::parseXSSProtectionHeader):
* Source/WebCore/platform/text/SegmentedString.cpp:
(WebCore::SegmentedString::advancePastSlowCase):
* Source/WebCore/platform/text/SegmentedString.h:
(WebCore::SegmentedString::advancePast):
(WebCore::SegmentedString::advancePastLettersIgnoringASCIICase):
* Source/WebCore/rendering/LegacyInlineBox.cpp:
(WebCore::LegacyInlineBox::boxName const):
(WebCore::LegacyInlineBox::outputLineBox const):
* Source/WebCore/rendering/LegacyInlineBox.h:
* Source/WebCore/rendering/LegacyInlineFlowBox.cpp:
(WebCore::LegacyInlineFlowBox::boxName const):
* Source/WebCore/rendering/LegacyInlineFlowBox.h:
* Source/WebCore/rendering/LegacyInlineTextBox.cpp:
(WebCore::LegacyInlineTextBox::boxName const):
(WebCore::LegacyInlineTextBox::outputLineBox const):
* Source/WebCore/rendering/LegacyInlineTextBox.h:
* Source/WebCore/rendering/LegacyRootInlineBox.cpp:
(WebCore::LegacyRootInlineBox::boxName const):
* Source/WebCore/rendering/LegacyRootInlineBox.h:
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::outputPaintOrderTreeRecursive):
(WebCore::showPaintOrderTree):
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::shouldDumpPropertyForLayer const):
* Source/WebCore/rendering/RenderLayerBacking.h:
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::shouldDumpPropertyForLayer const):
(WebCore::RenderLayerCompositor::updateBackingAndHierarchy):
(WebCore::RenderLayerCompositor::logLayerInfo):
(WebCore::compositingReasonToString):
(WebCore::RenderLayerCompositor::logOneReasonForCompositing):
* Source/WebCore/rendering/RenderLayerCompositor.h:
* Source/WebCore/rendering/RenderQuote.cpp:
(WebCore::quotesForLanguage):
* Source/WebCore/rendering/svg/SVGRenderTreeAsText.cpp:
(WebCore::operator<<):
(WebCore::writeNameValuePair):
(WebCore::writeNameAndQuotedValue):
(WebCore::writeIfNotEmpty):
(WebCore::writeIfNotDefault):
(WebCore::writeSVGPaintingResource):
(WebCore::writeSVGFillPaintingResource):
(WebCore::writeSVGStrokePaintingResource):
(WebCore::writeSVGPaintingFeatures):
(WebCore::writeSVGGraphicsElement):
(WebCore::writeRenderSVGTextBox):
(WebCore::writeSVGInlineTextBox):
(WebCore::writeStandardPrefix):
(WebCore::writeCommonGradientProperties):
(WebCore::writeSVGResourceContainer):
(WebCore::writeSVGContainer):
(WebCore::write):
(WebCore::writeSVGText):
(WebCore::writeSVGInlineText):
(WebCore::writeSVGImage):
(WebCore::writeSVGGradientStop):
(WebCore::writeResources):

Canonical link: https://commits.webkit.org/277913@main
